### PR TITLE
feat: add (w)steth unwrap/wrap on Sepolia Optimism

### PR DESCRIPTION
**Pull request by solidovic**
### Description

Added a (w)steth unwrap/wrap on Sepolia Optimism.

### Demo

image

### Checklist:

- [x] Checked the changes locally.
- [ ] Created / updated analytics events.
- [ ] Created / updated the technical documentation (README.md / [docs](https://docs.lido.fi/) / etc.).
- [ ] Affects / requires changes in other services (Matomo / Sentry / CloudFlare / etc.).

### CODE REVIEW

File: features/wsteth/wrap/wrap-form-controls/token-select-wrap.tsx [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
+    } else {
+      return [OPTION_STETH, OPTION_ETH];
+    }
+  }, [isDappActiveOnL2]);
 
   return (
     <TokenSelectHookForm
       disabled={isWalletConnected && !isDappActive}
-      options={OPTIONS}
+      options={options}
```
Comment (Jeday) [inline features/wsteth/wrap/wrap-form-controls/token-select-wrap.tsx:38]: There is no options in design

File: utils/get-etherscan-tx-link.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { getEtherscanTxLink as getEtherscanTxLinkSDK } from '@lido-sdk/helpers';
+import { CHAINS as CHAINS_SDK } from '@lido-sdk/constants';
+
+import { CHAINS } from 'consts/chains';
+
+export const getEtherscanTxLink = (
+  chainId: CHAINS | CHAINS_SDK,
+  hash: string,
+): string => {
+  if (chainId === CHAINS.OptimismSepolia) {
```
Comment (Jeday) [inline utils/get-etherscan-tx-link.ts:10]: You can get explorer url from viem chains object https://viem.sh/docs/chains/introduction

File: shared/hooks/use-dapp-status.ts [hunk 3]
```diff
@@ -19,24 +21,68 @@ export const useDappStatus = () => {
...
 
-  // TODO: rename isDappActive to isDappActiveOnL1
+  // TODO: rename to isAccountActive
   const isDappActive = useMemo(() => {
     if (!chainId) return false;
 
     return isWalletConnected && isSupportedChain;
   }, [chainId, isWalletConnected, isSupportedChain]);
 
+  const isAccountActiveOnL1 = useMemo(() => {
```
Comment (Jeday) [inline shared/hooks/use-dapp-status.ts:31]: All those memos can be unified into one via object.

File: shared/components/layout/header/components/chain-switcher/styles.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import styled, { css } from 'styled-components';
+import { SelectIcon } from '@lidofinance/lido-ui';
+
+export const SelectIconWrapper = styled.div`
+  position: relative;
+`;
+
+export const SelectIconStyled = styled(SelectIcon)`
```
Comment (Jeday) [inline shared/components/layout/header/components/chain-switcher/styles.tsx:8]: Height does not match design and rest of the header

File: providers/lido-sdk.tsx [hunk 5]
```diff
@@ -33,38 +31,38 @@ export const useLidoSDK = () => {
...
+    // @ts-expect-error: typing (rpcProvider + rpcUrls)
     const core = new LidoSDKCore({
       chainId,
       logMode: 'none',
-      rpcProvider: publicClient as any,
-      web3Provider: walletClient as any,
+      rpcProvider: publicClient,
+      web3Provider: walletClient,
       // viem client can be unavailable on ipfs+dev first renders
       rpcUrls: !publicClient && fallbackRpcUrl ? [fallbackRpcUrl] : undefined,
```
Comment (Jeday) [inline providers/lido-sdk.tsx:48]: This might not be actual anymore with usePublicClient

File: providers/lido-sdk.tsx [hunk 6]
```diff
@@ -1,24 +1,22 @@
...
 import { useGetRpcUrlByChainId } from 'config/rpc';
+import { useTokenTransferSubscription } from 'shared/hooks/use-balance';
 
 type LidoSDKContextValue = {
-  core: LidoSDKCore;
-  steth: LidoSDKstETH;
-  wsteth: LidoSDKwstETH;
-  l2Steth: LidoSDKL2Steth;
-  l2Wsteth: LidoSDKL2Wsteth;
+  lidoSDKCore: LidoSDKCore;
```
Comment (Jeday) [inline providers/lido-sdk.tsx:16]: let's use shorter names like in LidoSDK fields

File: providers/index.tsx [hunk 7]
```diff
@@ -21,18 +22,20 @@ export const Providers: FC<PropsWithChildren<ProvidersProps>> = ({
...
-          <LidoSDKProvider>
-            <IPFSInfoBoxStatusesProvider>
-              <InpageNavigationProvider>
-                <ModalProvider>{children}</ModalProvider>
-              </InpageNavigationProvider>
-            </IPFSInfoBoxStatusesProvider>
-          </LidoSDKProvider>
-        </Web3Provider>
-      </CookieThemeProvider>
+      <DappChainProvider>
```
Comment (Jeday) [inline providers/index.tsx:25]: Would it make sense to have it under web3provider and derive state from it

**Review by Jeday — CHANGES_REQUESTED** [review]

File: utils/get-etherscan-tx-link.ts (hunk 2 above)
Comment (solidovic) [reply utils/get-etherscan-tx-link.ts:10]: explorer

File: shared/hooks/use-dapp-status.ts (hunk 3 above)
Comment (solidovic) [reply shared/hooks/use-dapp-status.ts:31]: one big use memo

File: providers/lido-sdk.tsx (hunk 5 above)
Comment (solidovic) [reply providers/lido-sdk.tsx:48]: check removing rpcUrls

File: IPFS.json [hunk 8]
```diff
@@ -32,6 +32,15 @@
       "multiChainBanner": []
     }
   },
+  "10": {
```
[the remark this answers is not in the document]
Comment (solidovic) [reply IPFS.json:35]: done

File: shared/hooks/use-dapp-status.ts (hunk 3 above)
Comment (solidovic) [reply shared/hooks/use-dapp-status.ts:31]: done

File: shared/components/layout/header/components/chain-switcher/styles.tsx (hunk 4 above)
Comment (solidovic) [reply shared/components/layout/header/components/chain-switcher/styles.tsx:8]: fixed

File: providers/lido-sdk.tsx (hunk 6 above)
Comment (solidovic) [reply providers/lido-sdk.tsx:15]: refactored

File: providers/index.tsx (hunk 7 above)
Comment (solidovic) [reply providers/index.tsx:25]: Web3Provider implicitly depends on DappChainProvider

File: utils/get-etherscan-tx-link.ts (hunk 2 above)
Comment (solidovic) [reply utils/get-etherscan-tx-link.ts:10]: let's do it later

**Review by Jeday — DISMISSED** [review]

File: shared/hooks/use-wstETH-by-stETH-on-l2.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+export const useWstETHByStETHOnL2 = (wsteth: BigNumber | undefined) => {
+  const { l2 } = useLidoSDK();
+
+  const fetchSteth = async (amount: bigint) => {
+    return await l2.steth.convertToShares(amount);
+  };
+
+  const { data, error, isValidating, mutate } = useSWR(
+    wsteth ? [wsteth.toBigInt()] : null,
```
Comment (Jeday) [inline shared/hooks/use-wstETH-by-stETH-on-l2.ts:14]: Do this not fetch on null key? Needs rechek or comment.

File: shared/hooks/use-stETH-by-wstETH-on-l2.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+export const useStETHByWstETHOnL2 = (wsteth: BigNumber | undefined) => {
+  const { l2 } = useLidoSDK();
+
+  const fetchSteth = async (amount: bigint) => {
+    return await l2.steth.convertToSteth(amount);
+  };
+
+  const { data, error, isValidating, mutate } = useSWR(
+    wsteth ? [wsteth.toBigInt()] : null,
```
Comment (Jeday) [inline shared/hooks/use-stETH-by-wstETH-on-l2.ts:14]: same comment in useWstETHByStETHOnL2

File: shared/components/data-table-row-steth-by-wsteth/data-table-row-steth-by-wsteth.tsx [hunk 11]
```diff
@@ -15,20 +16,24 @@ type DataTableRowStethByWstethProps = {
 export const DataTableRowStethByWsteth = ({
   toSymbol = 'stETH',
 }: DataTableRowStethByWstethProps) => {
-  const { data, initialLoading } = useStethByWsteth(OneWsteth);
+  const { isDappActiveOnL2 } = useDappStatus();
+  const { data: dataOnL1, initialLoading: initialLoadingOnL1 } =
```
Comment (Jeday) [inline shared/components/data-table-row-steth-by-wsteth/data-table-row-steth-by-wsteth.tsx:20]: If typing is the same you could do 
```
const swrL2 = ...
const swrL1 = ...
const { data, initialLoading } = isL2 ? swrL2 : swrL1
```

File: features/wsteth/wrap/wrap-form/wrap-stats.tsx [hunk 12]
```diff
@@ -30,13 +31,18 @@ export const WrapFormStats = () => {
...
+  } = useDebouncedWstethBySteth(amount, isDappActiveOnL2);
 
   const {
-    data: oneWstethConverted,
-    initialLoading: isOneWstethConvertedLoading,
+    data: oneWstethConvertedOnL1,
+    initialLoading: isOneWstethConvertedLoadingOnL1,
   } = useWstethBySteth(oneSteth);
 
+  const {
```
Comment (Jeday) [inline features/wsteth/wrap/wrap-form/wrap-stats.tsx:41]: Same comment about shorting swr code

File: features/wsteth/unwrap/hooks/use-unwrap-tx-on-l2-approve.ts [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+export const useUnwrapTxOnL2Approve = ({ amount }: UseUnwrapTxApproveArgs) => {
+  const { isAccountActiveOnL2 } = useDappStatus();
+  const { lidoSDKCore, lidoSDKL2 } = useLidoSDK();
+  const [allowance, setAllowance] = useState<bigint | null>(null);
+  const [isApprovalNeededBeforeUnwrap, setIsApprovalNeededBeforeUnwrap] =
+    useState<boolean>(false);
+
+  const refetchAllowance = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (Jeday) [reply features/wsteth/unwrap/hooks/use-unwrap-tx-on-l2-approve.ts:19]: @solidovic  up

File: shared/hooks/use-wstETH-by-stETH-on-l2.ts (hunk 9 above)
Comment (solidovic) [reply shared/hooks/use-wstETH-by-stETH-on-l2.ts:14]: added comment 

' // if key is null, SWR will not fetch any data.' 

https://swr.vercel.app/docs/conditional-fetching#conditional

File: shared/hooks/use-stETH-by-wstETH-on-l2.ts (hunk 10 above)
Comment (solidovic) [reply shared/hooks/use-stETH-by-wstETH-on-l2.ts:14]: added comment 

' // if key is null, SWR will not fetch any data.' 

https://swr.vercel.app/docs/conditional-fetching#conditional

File: features/wsteth/wrap/wrap-form/wrap-stats.tsx (hunk 12 above)
Comment (solidovic) [reply features/wsteth/wrap/wrap-form/wrap-stats.tsx:41]: refactored

File: shared/components/data-table-row-steth-by-wsteth/data-table-row-steth-by-wsteth.tsx (hunk 11 above)
Comment (solidovic) [reply shared/components/data-table-row-steth-by-wsteth/data-table-row-steth-by-wsteth.tsx:20]: refactored

**Review by Jeday — APPROVED** [review]

**Review by Jeday — APPROVED** [review]

**Review by Jeday — APPROVED** [review]

**Review by Jeday — APPROVED** [review]

**Review by Jeday — DISMISSED** [review]

**Review by Jeday — APPROVED** [review]

File: shared/wallet/lido-multichain-fallback/lido-multichain-fallback.tsx [hunk 14]
```diff
@@ -46,8 +50,11 @@ const getChainLogo = (chainId: LIDO_MULTICHAIN_CHAINS) => {
 export const LidoMultichainFallback: LidoMultichainFallbackComponent = (
   props,
 ) => {
-  const { chainId } = useAccount();
+  const { chainId: chainIdWagmin } = useAccount();
```
[the remark this answers is not in the document]
Comment (Jeday) [reply shared/wallet/lido-multichain-fallback/lido-multichain-fallback.tsx:53]: sorry it as late

File: features/rewards/features/top-card/top-card.tsx [hunk 15]
```diff
@@ -20,7 +24,17 @@ export const TopCard: FC = () => {
...
+        <Fallback
+          error={`Unsupported chain. Please switch to ${CHAINS[defaultChain]} in your wallet.`}
+        />
+      ) : isAccountActiveOnL2 ? (
+        <Fallback
+          error={`Unsupported chain. Please switch to ${CHAINS[defaultChain]} in your wallet.`}
+        />
+      ) : (
+        <Wallet />
+      )}
```
[the remark this answers is not in the document]
Comment (Jeday) [reply features/rewards/features/top-card/top-card.tsx:37]: you are right, initialy we had different texts  here

[the remark this answers is not in the document]
Comment (solidovic) [reply features/rewards/features/top-card/top-card.tsx:37]: specially made (for the 'isAccountActiveOnL2' case can be anther text). will fix it soon (not this iteration).

**Pull request merged**